kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -821,6 +824,14 @@ public class TypeSpec private constructor(
         require(isAbstract || ABSTRACT !in propertySpec.modifiers) {
           "non-abstract type $name cannot declare abstract property ${propertySpec.name}"
         }
+        if (propertySpec.contextParameters.isNotEmpty()) {
+          if (ABSTRACT !in kind.implicitPropertyModifiers(modifiers) + propertySpec.modifiers) {
```

Comment:
This looks nice, however it would add two sets which is an overhead. Have you considered !in or !in approach to avoid set merging?

Reply:
Thanks! I didn’t pay much attention to performance considerations.

In this snippet, another improvement might be possible in terms of allocations. The call to `implicitPropertyModifiers` doesn’t seem necessary and could potentially be replaced with `defaultImplicitPropertyModifiers`. However, I’m not familiar enough with the internals to confidently say whether it makes sense to access this property directly.